server/src/test/java/org/opensearch/gateway/PrimaryShardBatchAllocatorTests.java

```diff
@@ -158,6 +163,46 @@ public void testMakeAllocationDecisionDataFetchedMultipleShards() {
         }
     }
 
+    public void testAllocateUnassignedBatchThrottlingAllocationDeciderNotHonoured() {
+        ClusterSettings clusterSettings = new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        AllocationDeciders allocationDeciders = randomAllocationDeciders(Settings.builder()
+            .put(ThrottlingAllocationDecider.CLUSTER_ROUTING_ALLOCATION_NODE_INITIAL_PRIMARIES_RECOVERIES_SETTING.getKey(), 1)
+            .build(), clusterSettings, random());
+        setUpShards(2);
+        final RoutingAllocation routingAllocation = routingAllocationWithMultiplePrimaries(
+            allocationDeciders,
+            CLUSTER_RECOVERED,
+            2,
+            0,
+            "allocId-0", "allocId-1"
+        );
+
+        List<ShardRouting> shards = new ArrayList<>();
+        for (ShardId shardId : shardsInBatch) {
+            ShardRouting shard = routingAllocation.routingTable().getIndicesRouting().get("test").shard(shardId.id()).primaryShard();
+            shards.add(shard);
+            batchAllocator.addShardData(
+                node1,
+                "allocId-" + shardId.id(),
+                shardId,
+                true,
+                new ReplicationCheckpoint(shardId, 20, 101, 1, Codec.getDefault().getName()),
+                null
+            );
+        }
+
+        allocateAllUnassignedBatch(routingAllocation);
+
+        // Verify the throttling decider was not throttled, recovering shards on node greater than initial concurrent recovery setting
+        assertFalse(routingAllocation.routingNodes().getInitialPrimariesIncomingRecoveries(node1.getId())
+            < 1);
+        List<ShardRouting> initializingShards = routingAllocation.routingNodes().shardsWithState(ShardRoutingState.INITIALIZING);
+        assertEquals(2, initializingShards.size());
```

Comment:
Why is this 2 ? Shouldn't it be one initialising shard on node1 only ?

Reply:
Updated the test in the latest revision.